com.unity.shadergraph/Editor/Data/Graphs/AbstractShaderProperty.cs

```diff
@@ -7,6 +7,36 @@ namespace UnityEditor.ShaderGraph.Internal
     [Serializable]
     public abstract class AbstractShaderProperty : ShaderInput
     {
+
+        public virtual int latestVersion { get; } = 0;
+        public virtual int version { get; protected set; } = 0;
```

Comment:
Should the version just be a private serialized variable at this level, rather than at the individual property type level? I feel like it should be. Get should still be public, and set should still be protected, but may not need to be virtual then. Latest version should probably stay virtual to allow each property to express its latest version